app/lib/search/token_index.dart

```diff
@@ -183,6 +184,33 @@ class ScorePool<K> {
   }
 }
 
+/// A reusable pool for [BitArray] instances to spare some memory allocation.
+class BitArrayPool<K> {
+  final int _length;
+  final _pool = <BitArray>[];
+
+  BitArrayPool(this._length);
+
+  BitArray _acquireAllSet() {
+    final array = _pool.isNotEmpty ? _pool.removeLast() : BitArray(_length);
+    array.setRange(0, _length);
```

Comment:
I guess you could do:
```suggestion
    final array = _pool.isNotEmpty ? _pool.removeLast()..setRange(0, _length) : BitArray(_length);
```

Reply:
Both need the `setRange` (which is probably a bad name, but a counter-operation of `clearRange` and sets (or clears) all the bits within the given range.

Reply:
Ah, sorry, thought the constructor would create a zeroed instance.